onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3018,23 +2985,39 @@ def aten_kthvalue(
 @torch_op("aten::layer_norm", trace_only=True)
 def aten_layer_norm(
     input: TReal,
-    normalized_shape: Sequence[int],
+    normalized_shape: INT64,
     weight: Optional[TReal] = None,
     bias: Optional[TReal] = None,
     eps: float = 1e-05,
 ) -> TReal:
     """layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True) -> Tensor"""
 
-    axes_list = [-i for i in range(len(normalized_shape), 0, -1)]
-    start_axis = axes_list[0]
-    if not op.OptionalHasElement(weight):
+    # trace_only to use Python to obtain start_axis
+    start_axis = -len(normalized_shape)
```

Comment:
This always confuses me I guess. As the ATen signature INT64 is not supported with len(), but it's actually not INT64 with trace_only.

Reply:
Should we annotate it following regular Python style when it's trace_only? And I think moving all preprocessing out is also important to align the ATen signature with what actually happends inside the ONNXFunction.

Reply:
It is INT64, or a TorchScriptTensor. I plan to reconcile the type system and update the INT64 to be a Tensor in the future

Reply:
> Should we annotate it following regular Python style when it's trace_only? And I think moving all preprocessing out is also important to align the ATen signature with what actually happends inside the ONNXFunction.

I think should update all signatures to Tensor when we fix typing in onnxscript. This way eager and script use the same type e.g.: `Tensor[INT64]`

Reply:
Ah, I see, and INT64 is what we define in tensor.py.